GitUI/HelperDialogs/FormResetAnotherBranch.cs

```diff
@@ -61,9 +62,28 @@ private IGitRef[] GetLocalBranchesWithoutCurrent()
             var currentBranch = Module.GetSelectedBranch();
             var isDetachedHead = currentBranch == DetachedHeadParser.DetachedBranch;
 
-            return Module.GetRefs(false)
+            IGitRef[] refs = Module.GetRefs(false)
                 .Where(r => r.IsHead && (isDetachedHead || r.LocalName != currentBranch))
                 .ToArray();
+
+            var remotes = _revision.Refs.Where(r => r.IsRemote).ToList();
+            if (remotes.Count == 0)
+            {
+                return refs;
+            }
+
+            var refsOrdered = new List<IGitRef>();
+            foreach (var remote in remotes)
+            {
+                IGitRef localBranch = refs.FirstOrDefault(r => r.MergeWith == remote.LocalName && r.TrackingRemote == remote.Remote);
+                if (localBranch != null)
+                {
+                    refsOrdered.Add(localBranch);
+                }
+            }
+
+            refsOrdered.AddRange(refs.Where(r => !refsOrdered.Contains(r)));
+            return refsOrdered.ToArray();
```

Comment:
Is there some git command we can execute to get the result in one go, and avoid multiple allocations and iterations?

Reply:
Not what I see.
However, the number of branches for a commit is normally limited and this command is not done so often.